### PR TITLE
remove android and iOS builds from CI

### PR DESCRIPTION
**Pull request by pmeier**
The android and iOS builds have been unmaintained for a long time now. As discussed in the latest Nova sync, there is no need to keep CI jobs running and we can kill them.

This PR also purges all android and iOS functionality and mentions from the repository. I don't see why we want to keep it if we not even have CI running for it, but no strong opinion. Happy to revert if needed.

cc @seemethere

### CODE REVIEW

**Review by seemethere — APPROVED** [review]
LGTM thanks for the quick turnaround on this!

**Comment by pmeier** [comment]
C source lint failure is valid. The rest is bogus.

**Review by osalpekar — APPROVED** [review]

File: torchvision/csrc/ops/quantized/cpu/qroi_align_kernel.cpp [hunk 1]
```diff
@@ -8,6 +8,9 @@ namespace ops {
 
 namespace {
 
+// FIXME: android is no longer supported and thus the comments below can be
+// acted on
+
```
Comment (NicolasHug) [inline torchvision/csrc/ops/quantized/cpu/qroi_align_kernel.cpp:13]: Not sure we want to remove that comment - it gives context on why the FIXME hasn't been done yet

**Comment by NicolasHug** [comment]
I confirmed with product team that:

- it's OK to remove the CI jobs
- it's OK to deprecate and stop support of the demos
- it's preferable to keep the android and ios folders around, with a note about their deprecated / unmaintained status

File: torchvision/csrc/ops/quantized/cpu/qroi_align_kernel.cpp (hunk 1 above)
Comment (pmeier) [reply torchvision/csrc/ops/quantized/cpu/qroi_align_kernel.cpp]: I've removed this again since we aren't purging mobile anyway now.

**Comment by pmeier** [comment]
For completion

> So far we only had to update gradle versions and stuff like that

is not the full story. It is right that we needed to only do that, but we were heavily relying on the CircleCI image for Android builds. This is no longer available through Nova. In #7410 I sunk significant time into getting it to work. Ultimately I did, but I don't know if the build is correct or even works.

**Pull request merged**